modules/kubernetes-addons/cluster-autoscaler/data.tf

```diff
@@ -10,9 +10,31 @@ data "aws_iam_policy_document" "cluster_autoscaler" {
       "autoscaling:DescribeLaunchConfigurations",
       "autoscaling:DescribeTags",
       "ec2:DescribeInstanceTypes",
-      "ec2:DescribeLaunchTemplateVersions",
+      "ec2:DescribeLaunchTemplateVersions"
+    ]
+  }
+
+  statement {
+    sid       = ""
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
       "autoscaling:SetDesiredCapacity",
-      "autoscaling:TerminateInstanceInAutoScalingGroup"
+      "autoscaling:TerminateInstanceInAutoScalingGroup",
+      "ec2:DescribeInstanceTypes",
+      "eks:DescribeNodegroup"
     ]
+    condition {
+      test     = "StringEquals"
+      variable = "autoscaling:ResourceTag/k8s.io/cluster-autoscaler/${var.addon_context.eks_cluster_id}"
+      values   = ["owned"]
+    }
+    condition {
```

Comment:
I think we can leave off this second test condition, the first one will override in both cases where cluster autoscaler is enabled or disabled

Reply:
Apologies, my brain is not on it today and I saw the first condition in the snippet highlight. 

Completely agree. Removed, as suggested.